### PR TITLE
Add explanatory comments to the LSP implementation

### PR DESCRIPTION
**Pull request by david072**
This adds some explanatory comments to the LSP implementation to help readability as requested by @soni801. However, I am not sure how to properly document this code. I would appreciate any feedback about sections that need more explaining, unnecessary/self-explanatory comments, etc.

### CODE REVIEW

**Comment by soni801** [comment]
This helps a lot!

I just skimmed through the comments, and I have two (minor) requests:
- Could document the tokenizer functions that aren't documented now?
- Could you maybe add some more inline comments to the `parse()` and `parseToolsField()` functions? I find them quite hard to follow hehe

Other than that, really happy with this! :-)

**Comment by david072** [comment]
Alright @soni801 I have added some more comments to `parse()` and `parseToolsField()` functions, as well as to the tokenizer's functions, and tried to simplify some code. I agree, the functions are very long and hard to understand, it might be possible to simplify them further in certain places or split them up into smaller functions.

I also included a little fix/refactor for path arguments (see `start` or `rngmanip`) that makes the code simpler and correctly handles paths after `rngmanip`. If you'd rather like that in a separate PR, let me know and I will try my best to bend the commit history to do that ;).

Hope this helps!
